### PR TITLE
inardini - experiments cuj2 notebook release

### PR DESCRIPTION
**Pull request by inardini**
A PR to release the official experiments notebook - CUJ2.

If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb [hunk 1]
```diff
@@ -0,0 +1,1229 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Below some of dataset records. -&gt; Below are example records from the dataset:</p><p> </p><p>you will learn how  -&gt; you learn how</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>move pip installs into the cell above (single cell for installs)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Use the notebook_template version for setting BUCKET and REGION</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Do you mean "Get the data"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>artefact -&gt; artifact</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Then -&gt; Next,</p><p> </p><p>covert -&gt; convert</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>I suggest more details about the Artifacts you are creating</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>you will train multinomial Naive Bayes pipeline.  -&gt; you train a multinomial Naive Bayes pipeline. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Below the training module. -&gt; Next, you create the training module.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Then -&gt; Next,</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (andrewferlitsch) [inline notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Use the method:</p><p>delete_bucket = False</p><p> </p><p>if delete_bucket or os.getenv("IS_TESTING"):</p><p>    ! gsutil -m rm -r n$BUCKET_URI</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='65'/>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>Correct. Change accordingly</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>I add a more explicit phase</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>done</p>

Comment (inardini) [reply notebooks/official/experiments/build_model_experimentation_lineage_with_prebuild_code.ipynb:1]: <p>done</p>

**Review by andrewferlitsch — APPROVED** [review]

**Comment by andrewferlitsch** [comment]
Failed on import:

 File /usr/local/lib/python3.9/site-packages/pandas/__init__.py:22, in <module>
Step #3:      19 del hard_dependencies, dependency, missing_dependencies
Step #3:      21 # numpy compat
Step #3: ---> 22 from pandas.compat import (
Step #3:      23     np_version_under1p18 as _np_version_under1p18,
Step #3:      24     is_numpy_dev as _is_numpy_dev,
Step #3:      25 )
Step #3:      27 try:
Step #3:      28     from pandas._libs import hashtable as _hashtable, lib as _lib, tslib as _tslib
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/compat/__init__.py:15, in <module>
Step #3:      12 import warnings
Step #3:      14 from pandas._typing import F
Step #3: ---> 15 from pandas.compat.numpy import (
Step #3:      16     is_numpy_dev,
Step #3:      17     np_array_datetime64_compat,
Step #3:      18     np_datetime64_compat,
Step #3:      19     np_version_under1p18,
Step #3:      20     np_version_under1p19,
Step #3:      21     np_version_under1p20,
Step #3:      22 )
Step #3:      23 from pandas.compat.pyarrow import (
Step #3:      24     pa_version_under1p0,
Step #3:      25     pa_version_under2p0,
Step #3:      26     pa_version_under3p0,
Step #3:      27     pa_version_under4p0,
Step #3:      28 )
Step #3:      30 PY38 = sys.version_info >= (3, 8)
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/compat/numpy/__init__.py:7, in <module>
Step #3:       3 import re
Step #3:       5 import numpy as np
Step #3: ----> 7 from pandas.util.version import Version
Step #3:       9 # numpy versioning
Step #3:      10 _np_version = np.__version__
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/util/__init__.py:1, in <module>
Step #3: ----> 1 from pandas.util._decorators import (  # noqa
Step #3:       2     Appender,
Step #3:       3     Substitution,
Step #3:       4     cache_readonly,
Step #3:       5 )
Step #3:       7 from pandas.core.util.hashing import (  # noqa
Step #3:       8     hash_array,
Step #3:       9     hash_pandas_object,
Step #3:      10 )
Step #3:      13 def __getattr__(name):
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/util/_decorators.py:14, in <module>
Step #3:       6 from typing import (
Step #3:       7     Any,
Step #3:       8     Callable,
Step #3:       9     Mapping,
Step #3:      10     cast,
Step #3:      11 )
Step #3:      12 import warnings
Step #3: ---> 14 from pandas._libs.properties import cache_readonly  # noqa
Step #3:      15 from pandas._typing import F
Step #3:      18 def deprecate(
Step #3:      19     name: str,
Step #3:      20     alternative: Callable[..., Any],
Step #3:    (...)
Step #3:      25     msg: str | None = None,
Step #3:      26 ) -> Callable[[F], F]:
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/_libs/__init__.py:14, in <module>
Step #3:       1 __all__ = [
Step #3:       2     "NaT",
Step #3:       3     "NaTType",
Step #3:    (...)
Step #3:       9     "Interval",
Step #3:      10 ]
Step #3:      13 from pandas._libs.interval import Interval
Step #3: ---> 14 from pandas._libs.tslibs import (
Step #3:      15     NaT,
Step #3:      16     NaTType,
Step #3:      17     OutOfBoundsDatetime,
Step #3:      18     Period,
Step #3:      19     Timedelta,
Step #3:      20     Timestamp,
Step #3:      21     iNaT,
Step #3:      22 )
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/_libs/tslibs/__init__.py:31, in <module>
Step #3:       1 __all__ = [
Step #3:       2     "dtypes",
Step #3:       3     "localize_pydatetime",
Step #3:    (...)
Step #3:      27     "tz_compare",
Step #3:      28 ]
Step #3:      30 from pandas._libs.tslibs import dtypes
Step #3: ---> 31 from pandas._libs.tslibs.conversion import (
Step #3:      32     OutOfBoundsTimedelta,
Step #3:      33     localize_pydatetime,
Step #3:      34 )
Step #3:      35 from pandas._libs.tslibs.dtypes import Resolution
Step #3:      36 from pandas._libs.tslibs.nattype import (
Step #3:      37     NaT,
Step #3:      38     NaTType,
Step #3:    (...)
Step #3:      41     nat_strings,
Step #3:      42 )
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/_libs/tslibs/conversion.pyx:63, in init pandas._libs.tslibs.conversion()
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/_libs/tslibs/parsing.pyx:1, in init pandas._libs.tslibs.parsing()
Step #3: 
Step #3: File /usr/local/lib/python3.9/site-packages/pandas/_libs/tslibs/offsets.pyx:21, in init pandas._libs.tslibs.offsets()
Step #3: 
Step #3: ModuleNotFoundError: No module named 'dateutil.easter'
Step #3: 
Finished Step #3

**Comment by kweinmeister** [comment]
Try extra pip install of `python-dateutil`?

**Comment by andrewferlitsch** [comment]
PR test ran 6 notebooks. One of the failures appears to be this notebook.

**Comment by kweinmeister** [comment]
@inardini the regular notebook passed tests. The debug one did not.

```ModuleNotFoundError: No module named 'dateutil.easter'```

Maybe it needs an extra pip install of `python-dateutil`

**Pull request merged**